### PR TITLE
Draft PR for test

### PR DESCRIPTION
**Pull request by trietmnj**

### CODE REVIEW

File: structureprovider/nsi-structure-provider.go [hunk 1]
```diff
@@ -47,13 +47,15 @@ func InitNSISP() nsiStreamProvider {
 	otp := structures.JsonOccupancyTypeProvider{}
 	otp.InitDefault()
 
+	// TODO probably don't hard code a possibly changing url
```
Comment (HenryGeorgist) [inline structureprovider/nsi-structure-provider.go:50]: we should use the HEC FWLINK url to obscure the location

File: structureprovider/factory.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+// StructureProvider can be a gpkDataSet, nsiStreamProvider, or shpDataSet
+type StructureProvider interface {
+	// TODO regulate more common methods
+	ByFips(fipscode string, sp consequences.StreamProcessor)
+	ByBbox(bbox geography.BBox, sp consequences.StreamProcessor)
+}
+
+type StructureProviderInfo struct {
+	StructureProviderType *string // nsi or local
```
Comment (HenryGeorgist) [inline structureprovider/factory.go:19]: can we make this an enum?

File: structureprovider/factory.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	StructureFilePath     *string
```
Comment (HenryGeorgist) [inline structureprovider/factory.go:20]: This is also optional

**Review by HenryGeorgist — APPROVED** [review]

File: structureprovider/nsi-structure-provider.go (hunk 1 above)
Comment (trietmnj) [reply structureprovider/nsi-structure-provider.go:50]: Probably should get it from a .gitignore configuration file and wipe that info from this public repo

Comment (HenryGeorgist) [reply structureprovider/nsi-structure-provider.go:50]: fair enough

File: structureprovider/factory.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/USACE/go-consequences/consequences"
+	"github.com/USACE/go-consequences/geography"
+)
+
+const (
```
Comment (trietmnj) [inline structureprovider/factory.go:11]: How about string enums? It's easier to generate custom validation errors

File: structureprovider/factory.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	Provider_NSI   string = "Provider_NSI"
+	Provider_Local        = "Provider_Local"
+)
```
Comment (trietmnj) [inline structureprovider/factory.go:14]: How about string enums? It's easier to generate custom errors

File: structureprovider/factory.go (hunk 2 above)
Comment (trietmnj) [reply structureprovider/factory.go]: How about string enums? See below

File: structureprovider/factory.go (hunk 5 above)
Comment (HenryGeorgist) [reply structureprovider/factory.go:14]: that is great

**Pull request merged**